.github/workflows/publish.yml

```diff
@@ -11,34 +11,78 @@ on:
     paths:
       - "src/**"
       - ".github/workflows/publish.yml"
+  repository_dispatch:
+    types: [create-release]
+    client_payload:
+      release_tag:
+        required: true
 
 jobs:
+  create_release:
+    if: ${{ github.event_name == 'repository_dispatch' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: Brightspace/third-party-actions@actions/checkout
+      - name: Create release
+        env:
+          RELEASE_TAG: ${{ github.event.client_payload.release_tag }}
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          gh release create "$RELEASE_TAG" --draft --generate-notes --title "Release $RELEASE_TAG"
+
   build:
+    needs: create_release
+    if: ${{ !failure() }}
     strategy:
       matrix:
         include:
           - machine: windows-latest
             platform: win
+            file_name: bmx.exe
           - machine: macos-latest
             platform: osx
+            file_name: bmx
     runs-on: ${{ matrix.machine }}
     timeout-minutes: 10
+    env:
+      PLATFORM: ${{ matrix.platform }}
     steps:
       - name: checkout
         uses: Brightspace/third-party-actions@actions/checkout
+      - name: install 7zip
+        if: matrix.platform == 'win'
+        run: choco install 7zip -y
+        shell: bash
```

Comment:
powershell should meet our zipping needs in all OSes
https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.archive/compress-archive?view=powershell-7.3